### PR TITLE
Rust 1.80.0

### PR DESCRIPTION
**Pull request by squadgazzz**
Updates Rust to the latest version.

### CODE REVIEW

File: crates/shared/src/sources/balancer_v2/pool_init.rs [hunk 1]
```diff
@@ -22,6 +22,7 @@ pub trait PoolInitializing: Send + Sync {
 ///
 /// This can be used to index all pools from events instead of relying on the
 /// Balancer subgraph for example.
+#[allow(dead_code)]
```
Comment (Mateo-mro) [inline crates/shared/src/sources/balancer_v2/pool_init.rs:25]: They get better and better detecting dead code! :)

**Review by Mateo-mro — APPROVED** [review]

File: crates/shared/src/sources/balancer_v2/pool_init.rs (hunk 1 above)
Comment (squadgazzz) [reply crates/shared/src/sources/balancer_v2/pool_init.rs:25]: I am not sure about plans for this struct, though. Maybe it can be removed.

Comment (fleupold) [reply crates/shared/src/sources/balancer_v2/pool_init.rs:25]: Let's remove dead code if we find it (or condition it on the right feature/cfg that uses it). Chances are it was just forgotten.

**Review by fleupold — APPROVED** [review]
👏

**Pull request merged**